Mathlib/Algebra/Module/LinearMap/End.lean

```diff
@@ -203,6 +203,11 @@ theorem surjective_of_iterate_surjective {n : ℕ} (hn : n ≠ 0) (h : Surjectiv
   exact Surjective.of_comp h
 #align linear_map.surjective_of_iterate_surjective LinearMap.surjective_of_iterate_surjective
 
+lemma iterate_apply_eq_zero_of_le {m n : ℕ} {x : M} (hmn : m ≤ n) (hf : (f' ^ m) x = 0) :
+    (f' ^ n) x = 0 := by
+  obtain ⟨k, rfl⟩ := Nat.exists_eq_add_of_le hmn
+  rw [add_comm _ k, pow_add, LinearMap.mul_apply, hf, map_zero]
+
```

Comment:
This is exactly `LinearMap.pow_map_zero_of_le`.
```suggestion
```
I think the poor naming in this corner of the library contributed to not finding `LinearMap.pow_map_zero_of_le`. (There are a bunch of lemmas here that are about `pow` but whose name incorrectly says `iterate` and the lemma `LinearMap.pow_map_zero_of_le` should really be named `LinearMap.pow_apply_zero_of_le`.)

Reply:
Ok, I'll just remove this. These changes should be done in a follow-up PR.